coinstacks/bitcoin/ingester/src/parseTx/index.ts

```diff
@@ -0,0 +1,55 @@
+import { BigNumber } from 'bignumber.js'
+import { Tx, Vin, Vout } from '@shapeshiftoss/blockbook'
+import { TxTransfer, Token } from '@shapeshiftoss/common-ingester'
+import { BTCParseTx } from '../types'
+
+// keep track of all individual tx components and add up the total value transferred
+const aggregateTransfer = (transfer: TxTransfer, value: string, token?: Token): TxTransfer => {
+  if (transfer) {
+    transfer.totalValue = new BigNumber(transfer.totalValue).plus(new BigNumber(value)).toString(10)
+    transfer.components.push({ value: value })
+  } else {
+    transfer = { totalValue: value, components: [{ value: value }] }
+  }
+
+  return { ...transfer, token }
+}
+
+export const parseTx = async (tx: Tx, address: string): Promise<BTCParseTx> => {
+  const pTx: BTCParseTx = {
+    ...tx,
+    address,
+    receive: {},
+    send: {},
+  }
+
+  // todo - assumptions:
+  // - all inputs spend, no change address
+  // get send amount
+  tx.vin.forEach((vin: Vin) => {
+    if (vin.isAddress === true && vin.addresses?.[0] === pTx.address) {
+      const sendValue = new BigNumber(vin.value ?? 0)
```

Comment:
Thinking about it more, we probably don't even need the `?? 0` since it will either convert to `NaN` which we are checking for, or be a valid number. This would also imply we don't need the `|| 0` on line 33 since it is within the if and confirmed as a number > 0.

Reply:
got errors when removed so left them in

```
src/parseTx/index.ts(29,39): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'Value'.
```